test/utils/ruleset-test-harness.ts

```diff
@@ -139,16 +136,17 @@ export const getRulesetTestEnvironments = (): RulesetTestEnvironment[] => {
     return [chromium, firefox, webkit].map((browserType) => {
       // Reuse a single browser instance for each browser type across all tests.
       // This reduces browser test run time by about half.
-      const browserPromise = browserType.launch();
+      let browser: Browser | undefined;
       return {
         name: browserType.name(),
         createTestHarness: async (rules: DataLayerRule[], dataLayer: any) => {
-          const testHarness = new BrowserTestHarness(browserPromise);
+          browser = browser || await browserType.launch();
```

Comment:
If the condition were flipped we would we default to the non-nullable 'browser' object. Why does it need to be first?

Reply:
I'm not sure I understand the question. The intent here is to only launch a given browser once for a test environment instance, and this leans into the fact that we're not currently parallelizing tests so the first test harness created should pay the cost. I'm okay with the compromise on parallelization since, in my testing, running these tests through Mocha with the `--parallel` switch significantly increases test run time.

Reply:
I was wondering what the difference of "browser || browser launch" vs "browser launch || browser" in the code statement

Reply:
This cleared it up thank you!

Reply:
Ah, great, thanks! Yeah, once `browser` is defined as it'll be after the first `await` we'll no longer try to launch another browser instance for the same browser type.

Reply:
Yeah, the scope in JS threw me off 😵